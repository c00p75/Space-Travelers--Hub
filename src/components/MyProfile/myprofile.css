.summary {
  max-height: 63vh;
  overflow: auto;
}

.summary::-webkit-scrollbar {
  width: 10px;
  background-color: rgb(255, 255, 255);
  border-bottom-right-radius: 5px;
}

.summary::-webkit-scrollbar-thumb {
  background: gray;
  border-radius: 10px;
}

@media screen and (max-width: 576px) {
  .summary {
    max-height: 100%;
  }
}
