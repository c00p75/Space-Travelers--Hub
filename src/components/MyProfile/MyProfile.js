import { useSelector, useDispatch } from 'react-redux';
import { joinMission } from '../../redux/missions/missions';
import { booking } from '../../redux/dragons.js/dragons';
import { reservation } from '../../redux/rockets/rockets';
import './myprofile.css';

const MyProfile = () => {
  const dispatch = useDispatch();
  const missionsStore = useSelector((state) => state.missions);
  const allRockets = useSelector((state) => state);
  const allDragons = useSelector((state) => state.dragons);
  const { missions } = missionsStore;
  const { rockets } = allRockets;
  const { dragons } = allDragons;
  const myMissions = missions.filter((mission) => mission.joined === true);
  const myRockets = rockets.filter((rocket) => rocket.reserved === true);
  const myDragons = dragons.filter((dragon) => dragon.booked === true);
  return (
    <div className="container-fluid">
      <div className="justify-content-between d-flex gap-5 gap-md-1 flex-column flex-md-row">
        <div className="col-md-4">
          <h1 className="p-2 bg-info">My Missions</h1>
          <div className="summary">
            {!myMissions.length ? 'No Missions Joined' : ' '}
            {myMissions.map((mission) => (
              <div className="border p-3 d-flex justify-content-between container-fluid" key={mission.id}>
                <div className="col-5">{mission.name}</div>
                <div>
                  <button
                    type="button"
                    className="btn btn-outline-danger"
                    onClick={() => dispatch(joinMission(mission))}
                  >
                    Leave Mission
                  </button>
                </div>
              </div>
            ))}
          </div>
        </div>
        <div className="col-md-4">
          <h1 className="p-2 bg-info">My Rockets</h1>
          <div>
            {!myRockets.length ? 'No Rockets Reserved' : ' '}
            {myRockets.map((rocket) => (
              <div className="border p-3 d-flex justify-content-between container-fluid" key={rocket.id}>
                <div className="col-5">{rocket.rocket_name}</div>
                <div>
                  <button type="button" className="btn btn-outline-secondary" onClick={() => dispatch(reservation(rocket.id))}>
                    Cancel Reservation
                  </button>
                </div>
              </div>
            ))}
          </div>
        </div>
        <div className="col-md-4">
          <h1 className="p-2 bg-info">My Dragons</h1>
          <div>
            {!myDragons.length ? 'No Dragons Booked' : ' '}
            {myDragons.map((dragon) => (
              <div className="border p-3 d-flex justify-content-between container-fluid" key={dragon.id}>
                <div>{dragon.name}</div>
                <div>
                  <button type="button" className="btn btn-outline-secondary" onClick={() => dispatch(booking(dragon.id))}>
                    Cancel Booking
                    {' '}
                  </button>
                </div>
              </div>
            ))}
          </div>
        </div>
      </div>
    </div>
  );
};

export default MyProfile;
