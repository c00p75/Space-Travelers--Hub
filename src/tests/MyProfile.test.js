import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MyProfile from '../components/MyProfile/MyProfile';

it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MyProfile />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
