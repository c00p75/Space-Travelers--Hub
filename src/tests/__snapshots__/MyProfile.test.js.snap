// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="justify-content-between d-flex gap-5 gap-md-1 flex-column flex-md-row"
  >
    <div
      className="col-md-4"
    >
      <h1
        className="p-2 bg-info"
      >
        My Missions
      </h1>
      <div
        className="summary"
      >
        No Missions Joined
      </div>
    </div>
    <div
      className="col-md-4"
    >
      <h1
        className="p-2 bg-info"
      >
        My Rockets
      </h1>
      <div>
        No Rockets Reserved
      </div>
    </div>
    <div
      className="col-md-4"
    >
      <h1
        className="p-2 bg-info"
      >
        My Dragons
      </h1>
      <div>
        No Dragons Booked
      </div>
    </div>
  </div>
</div>
`;
